Lib/urllib/parse.py

```diff
@@ -166,7 +166,10 @@ def hostname(self):
     def port(self):
         port = self._hostinfo[1]
         if port is not None:
-            port = int(port, 10)
+            try:
+                port = int(port, 10)
+            except ValueError:
+                raise ValueError("Port could not be cast to integer value") from None
```

Comment:
In order to make the diagnostic message more useful, I would suggest including the bad port value here.